pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/QueryQuotaClusterIntegrationTest.java

```diff
@@ -234,30 +235,44 @@ public void testApplicationAndDatabaseQueryQuotaWithTableQueryQuotaWithExtraBrok
    * Then runs the query load with double the max rate and expects queries to fail due to quota breach.
    * @param maxRate max rate allowed by the quota
    */
-  void testQueryRate(float maxRate)
-      throws Exception {
+  void testQueryRate(int maxRate) {
     verifyQuotaUpdate(maxRate);
     runQueries(maxRate, false);
     //increase the qps and some of the queries should be throttled.
     runQueries(maxRate * 2, true);
   }
 
-  void testQueryRateOnBroker(float maxRate)
-      throws Exception {
+  void testQueryRateOnBroker(float maxRate) {
     verifyQuotaUpdate(maxRate);
     runQueriesOnBroker(maxRate, false);
     //increase the qps and some of the queries should be throttled.
     runQueriesOnBroker(maxRate * 2, true);
   }
 
+  private static void sleep(long end, double iterationsLeft) {
+    long duration = (long) Math.max(Math.ceil((end - System.currentTimeMillis()) / iterationsLeft), 0);
```

Comment:
We can read `System.currentTimeMillis()` only once.
Some javadoc can help explain the logic

Reply:
Added comments the the integration test method.